test/topology_random_failures/cluster_events.py

```diff
@@ -0,0 +1,624 @@
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from __future__ import annotations
+
+import os
+import asyncio
+import logging
+from shutil import rmtree
+from typing import TYPE_CHECKING
+
+from test.pylib.tablets import get_all_tablet_replicas
+from test.topology.util import get_coordinator_host, get_non_coordinator_host
+from test.topology_random_failures.error_injections import ERROR_INJECTIONS
+
+if TYPE_CHECKING:
```

Comment:
is this `if` needed?

Reply:
Here it's just a habit to divide a stuff used just for type checks only from the runtime stuff. It works together with `from __future__ import annotations`.

But in general, this construction is for avoiding circular imports sometimes required by type hints.

Little bit more here: https://vickiboykis.com/2023/12/11/why-if-type_checking/ (for example)

Also, see https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING

Reply:
If conditional is not strictly needed (it will work without it), I'd remove it. And here IIUC there are no circular imports so we can get rid of it

Reply:
Think of this as a good habit: if some typing stuff is not required for runtime then it should be excluded by this check.